### PR TITLE
save all & commit option now commits all files (fixed #80837)

### LINKED ISSUE #80837: save all and commit option only commits unsaved files
Version: 1.38.0
Commit: 3db7e09f3b61f915d03bbfa58e258d6eee843f35
Date: 2019-09-03T21:47:00.464Z
Electron: 4.2.10
Chrome: 69.0.3497.128
Node.js: 10.11.0
V8: 6.9.427.31-electron.0
OS: Darwin x64 18.0.0

Steps to Reproduce:

1 open multiple tabs and only save some of them.
2 commit and click `Save All & Commit`

Does this issue occur when all extensions are disabled?: Yes

### PR DESCRIPTION
**Pull request by asutosh97**
Fixes #80837

### CODE REVIEW

File: extensions/git/src/commands.ts [hunk 1]
```diff
@@ -1251,7 +1251,7 @@ export class CommandCenter {
 
 		if (promptToSaveFilesBeforeCommit !== 'never') {
 			let documents = workspace.textDocuments
-				.filter(d => !d.isUntitled && d.isDirty && isDescendant(repository.root, d.uri.fsPath));
+				.filter(d => !d.isUntitled && d.uri.scheme === 'file' && isDescendant(repository.root, d.uri.fsPath));
```
Comment (joaomoreno) [inline extensions/git/src/commands.ts]: No, I think the fix should rather be around line 1271... all files should be staged, not just the list of `documents`.

**Review by joaomoreno — CHANGES_REQUESTED** [review]

File: extensions/git/src/commands.ts (hunk 1 above)
Comment (asutosh97) [reply extensions/git/src/commands.ts]: But having `d.isDirty` is excluding the documents that are saved. That's why I removed it.

Comment (joaomoreno) [reply extensions/git/src/commands.ts]: Calling `add([])` will stage everything.

**Comment by joaomoreno** [comment]
Closing in favor of https://github.com/microsoft/vscode/pull/81609

**Pull request closed without merging**